carddav/client.go

```diff
@@ -429,3 +429,62 @@ func (c *Client) PutAddressObject(path string, card vcard.Card) (*AddressObject,
 	}
 	return ao, nil
 }
+
+// SyncCollection do a sync-collection operation on resource(path), it returns an array of modified/deleted/new
+// AddressObject, the next SyncToken and error
+func (c *Client) SyncCollection(path string, query *SyncQuery) (*SyncResponse, error) {
+	var limit *internal.Limit
+	if query.Limit > 0 {
+		limit = &internal.Limit{NResults: uint(query.Limit)}
+	}
+
+	propReq, err := encodeAddressPropReq(&query.DataRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	ms, err := c.ic.SyncCollection(path, query.SyncToken, internal.DepthOne, limit, propReq)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &SyncResponse{}
+	for _, resp := range ms.Responses {
+		p, err := resp.Path()
+		if err != nil {
```

Comment:
I think `Path` returns an error that verifies `internal.IsNotFound` when the status is 404.

Would be nice to actually test that this code works :)